db/crud.go

```diff
@@ -1165,6 +1165,26 @@ func (db *DatabaseCollectionWithUser) PutExistingRevWithBody(ctx context.Context
 
 }
 
+func (db *DatabaseCollectionWithUser) SyncFnDryrun(ctx context.Context, body Body, docID string) (*uint32, string, base.Set, channels.AccessMap, channels.AccessMap, error) {
+	doc := &Document{
+		ID:    docID,
+		_body: body,
+	}
+	// Make sure the user has permission to modify the document before confirming doc existence
+	mutableBody, metaMap, newRevID, err := db.prepareSyncFn(doc, doc)
+	if err != nil {
+		base.InfofCtx(ctx, base.KeyCRUD, "Failed to prepare to run sync function: %v", err)
+		return nil, "", nil, nil, nil, err
+	}
+
+	syncExpiry, oldBody, channelSet, access, roles, err := db.runSyncFn(ctx, doc, mutableBody, metaMap, newRevID)
```

Comment:
runSyncFn increments several stats (in getChannelsAndAccess and checkDocChannelsAndGrantsLimits). I suspect we don't want those stats to be updated for dry runs.  This might require a 'dry run' flag to be added to the underlying code - I think that's more straightforward than messing around with the database context used for diagnostic operations.